crowbar_framework/app/models/node.rb

```diff
@@ -1493,10 +1497,11 @@ def find(search)
       end
 
       if nodes.is_a?(Array) and nodes[2] != 0 and !nodes[0].nil?
+        roles = Hash[RoleObject.all.map.collect { |role| [role.name, role] }]
```

Comment:
Asking out loud: should we do this just if `search.nil?`?

Reply:
Good point. Added.